File: frontend/packages/admin/scripts/fix-mam-imports.py
import os

root = os.path.join(os.path.dirname(__file__), '..', 'src', 'mam')
repls = [
    ("from '@/api/", "from '@/mam/api/"),
    ('from "@/api/', 'from "@/mam/api/'),
    ("from '@/components/", "from '@/mam/components/"),
    ("from '@/composables/", "from '@/mam/composables/"),
    ("from '@/constants/", "from '@/mam/constants/"),
    ("from '@/utils/material", "from '@/mam/utils/material"),
]

for dirpath, _, files in os.walk(root):
    for f in files:
        if not f.endswith(('.ts', '.vue')):
            continue
        p = os.path.join(dirpath, f)
        with open(p, 'r', encoding='utf-8') as fh:
            content = fh.read()
        updated = content
        for old, new in repls:
            updated = updated.replace(old, new)
        if updated != content:
            with open(p, 'w', encoding='utf-8', newline='\n') as fh:
                fh.write(updated)

print('mam imports updated')

File: frontend/packages/admin/scripts/fix-mam-imports.mjs
import fs from 'fs'
import path from 'path'
import {fileURLToPath} from 'url'

const __dirname = path.dirname(fileURLToPath(import.meta.url))
const root = path.resolve(__dirname, '../src/mam')
const repls = [
    ["from '@/api/", "from '@/mam/api/"],
    ['from "@/api/', 'from "@/mam/api/'],
    ["from '@/components/", "from '@/mam/components/"],
    ["from '@/composables/", "from '@/mam/composables/"],
    ["from '@/constants/", "from '@/mam/constants/"],
    ["from '@/utils/material", "from '@/mam/utils/material"],
]

function walk(dir) {
    for (const name of fs.readdirSync(dir, {withFileTypes: true})) {
        const p = path.join(dir, name.name)
        if (name.isDirectory()) {
            walk(p)
            continue
        }
        if (!/\.(ts|vue)$/.test(name.name)) continue
        let content = fs.readFileSync(p, 'utf8')
        let updated = content
        for (const [old, neu] of repls) {
            updated = updated.split(old).join(neu)
        }
        if (updated !== content) {
            fs.writeFileSync(p, updated, 'utf8')
        }
    }
}

walk(root)
console.log('mam imports updated')
